streams/src/main/java/org/apache/kafka/streams/processor/internals/StreamThread.java

```diff
@@ -416,6 +419,10 @@ private void shutdownTasksAndState() {
         if (cleanRun && firstException.get() == null) {
             firstException.set(commitOffsets());
         }
+        // Close all task producers
+        if (exactlyOnceEnabled) {
```

Comment:
see comment on `StreamTask` do we need to have a guard both here and in `StreamTask#closeProducer`?